### DISCUSSION TITLE
Searchable() doesn't pre-populate with data

### DISCUSSION THREAD

**Discussion by pnofrc** [laravel/nova-issues#5945] [Q&A]
Hi, I'm using Laravel 10 and I need to use the searchable function inside a BelongsTo.

The searchable function works but it doesn't display the existing data. 

The composer packages I found are deprecated. Anybody got a solution for this?

Thanks,
Federico

**Comment by Epizefiri** [comment]
The goal would be something similar to this
![image](https://github.com/laravel/nova-issues/assets/10724326/0e270b8b-b120-4f6c-9838-412f6696c31f)

**Comment by ghost** [comment]
We also stumbled upon this issue. It would be great to have the ability to preload existing options, similar to what we have for Tag fields.

**Comment by 8devil3** [comment]
Please, it would be great to have a searchable field.

**Comment by philipbaginski** [comment]
If you do not need use ->showCreateRelationButton() in BelongsTo field, simply use Select field and set up like:

![Select](https://github.com/laravel/nova-issues/assets/59031869/83de7249-80e4-432f-9455-c94ad8d17615)

->searchable() inside Select field gives you search field, and ->options() will give you list of options based on the model.

**Reply by RealityGamin** [reply]
Thats a pretty interesting approach :O

**Reply by davidhemphill** [reply]
This can have performance implications, since that query will be evaluated every time Nova uses the resource. It's best to wrap the database call in a Closure so that it's only evaluated when needed.

**Reply by pnofrc** [reply]
For what I had to do, this is perfect. big up for @philipbaginski !
